src/imitation/algorithms/dagger.py

```diff
@@ -391,30 +469,167 @@ def save_trainer(self) -> Tuple[str, str]:
+        dataset aggregation stage is determined by the `rollout_round_min*` arguments.
+
+        During a BC update step, `BC.train()` is called to update the DAgger agent on
+        all data collected so far.
+
+        Args:
+            total_timesteps: The number of timesteps to train inside the environment.
+                In practice this is a lower bound, because the number of timesteps is
+                rounded up to finish the minimum number of episdoes or timesteps in the
+                last DAgger training round, and the environment timesteps are executed
+                in multiples of `self.venv.num_envs`.
+            rollout_round_min_episodes: The number of episodes the must be completed
+                completed before a dataset aggregation step ends.
+            rollout_round_min_timesteps: The number of environment timesteps that must
+                be completed before a dataset aggregation step ends. Also, that any
+                round will always train for at least `self.batch_size` timesteps,
+                because otherwise BC could fail to receive any batches.
+            bc_train_kwargs: Keyword arguments for calling `BC.train()`. If
+                the `log_rollouts_venv` key is not provided, then it is set to
+                `self.venv` by default. If neither of the `n_epochs` and `n_batches`
+                keys are provided, then `n_epochs` is set to `self.DEFAULT_N_EPOCHS`.
+        """
+        total_timestep_count = 0
+        round_num = 0
+
+        while total_timestep_count < total_timesteps:
+            collector = self.get_trajectory_collector()
+            round_episode_count = 0
+            round_timestep_count = 0
+
+            sample_until = rollout.make_sample_until(
+                min_timesteps=max(rollout_round_min_timesteps, self.batch_size),
+                min_episodes=rollout_round_min_episodes,
+            )
+
+            trajectories = rollout.generate_trajectories(
+                policy=self.expert_policy,
+                venv=collector,
+                sample_until=sample_until,
+                deterministic_policy=True,
+                rng=collector.rng,
+            )
+
+            for traj in trajectories:
+                _save_dagger_demo(traj, collector.save_dir)
+                logger.record_mean("dagger/mean_episode_reward", np.sum(traj.rews))
+                round_timestep_count += len(traj)
+                total_timestep_count += len(traj)
+
+            round_episode_count += len(trajectories)
+
+            logger.record("dagger/total_timesteps", total_timestep_count)
+            logger.record("dagger/round_num", round_num)
+            logger.record("dagger/round_episode_count", round_episode_count)
+            logger.record("dagger/round_timestep_count", round_timestep_count)
+
+            # TODO(shwang): BC starts looping Tensorboard
+            #   back to x=0 with each new call to BC.train(). Consider adding a
```

Comment:
I've fixed this Tensorboard reset problem locally because it was making DAgger runs annoying to analyze. Commit will come shortly